lua/go.lua

```diff
@@ -13,7 +13,9 @@ _GO_NVIM_CFG = {
   log_path = vim.fn.expand("$HOME") .. "/tmp/gonvim.log",
   lsp_cfg = false, -- true: apply go.nvim non-default gopls setup
   lsp_gofumpt = false, -- true: set default gofmt in gopls format to gofumpt
-  lsp_on_attach = false, -- provides a on_attach function to gopls, will use go.nvim on_attach if nil
+  lsp_on_attach = true, -- true: use on_attach function specified in go/lsp.lua,
```

Comment:
In `go.lua`, there is this:

```
  if _GO_NVIM_CFG.lsp_cfg then
    require("go.lsp").setup()
 ```
 
 So even if `lsp_on_attach` is set to `true`, if `lsp_cfg` is `false`, this will never get set.

Reply:
And in the README, lsp_on_attach is specified as true: https://github.com/ray-x/go.nvim/blob/c063817d05f0c067111b037588a49b80d932d3a5/README.md#configuration

Reply:
require("go.lsp").setup()
does not mean hook the on_attach function
on_attach setup is here:
```
  if _GO_NVIM_CFG.lsp_on_attach then
    if _GO_NVIM_CFG.lsp_on_attach == true then
      gopls.on_attach = on_attach
    else
      gopls.on_attach = _GO_NVIM_CFG.lsp_on_attach
    end
  end
```
if it set to `false` default on_attach will not be attached. Should we enforce it been attached?

Reply:
Right now you have to do:

```
require('go').setup({lsp_cfg = true, lsp_on_attach = true})
```

You **must** have both of them set, otherwise they on_attach is not set.
This is confusing and not obvious from the docs.

I'm not sure how you want to structure this.

Do you want to be able to do:

```
require('go').setup({lsp_cfg = false, lsp_on_attach = true})
```

This is the problem when you have options that depend on each other.

Reply:
let's do this
lsp_on_attach can only be a function or nil
if lsp_on_attach is a function the function will be assigned to on_attach of gopls. If it is nil will use the default function.

Reply:
That sounds ok.  And the default will be `lsp_on_attach = nil`?

Reply:
also, is this a valid combination?

`require('go').setup({lsp_cfg = false, lsp_on_attach = my_attach_fn})`

Reply:
Default will be `nil` means use default on_attach function when lsp_cfg is true
if it is false. lsp_on_attach will not be used as we are not loading gopls from go.nvim

Reply:
Thanks for the clarification.
So based on what you are saying:

1.  This will do nothing, because it won't call the function in `go/lsp.lua`:

```
require('go').setup({lsp_cfg = false, lsp_on_attach = my_attach_fn})
```

2.  This will call the function in `go/lsp.lua`, and use the `on_attach` function defined in `go/lsp.lua`:
```
require('go').setup({lsp_cfg = true})
```

3.  This will call the function in `go/lsp.lua`, and use the `on_attach` function defined in `go/lsp.lua`:
```
require('go').setup({lsp_cfg = true, lsp_on_attach = nil})
```

4.  This will call the function in `go/lsp.lua`, and use `my_attach_fn` instead of the `on_attach` function defined in `go/lsp.lua`:

```
require('go').setup({lsp_cfg = true, lsp_on_attach = my_attach_fn})
```

Is my understanding correct?